File: android_app/app/src/main/java/com/oceansentinels/app/domain/repository/AnalyticsRepository.kt
package com.oceansentinels.app.domain.repository

import com.oceansentinels.app.domain.model.*

/**
 * Repository interface for analytics operations
 */
interface AnalyticsRepository {
    
    /**
     * Get dashboard analytics overview
     */
    suspend fun getDashboardAnalytics(): Result<DashboardAnalytics>
    
    /**
     * Get incidents timeline for the specified number of days
     */
    suspend fun getIncidentsTimeline(days: Int = 30): Result<IncidentsTimeline>
    
    /**
     * Get incidents distribution by status, type, and urgency
     */
    suspend fun getIncidentsDistribution(): Result<IncidentsDistribution>
    
    /**
     * Get geographic analytics
     */
    suspend fun getGeographicAnalytics(): Result<GeographicAnalytics>
    
    /**
     * Get all analytics data at once
     */
    suspend fun getAllAnalytics(): Result<AnalyticsData>
}

File: IOS/OceanSentinels/Domain/Repositories/AnalyticsRepository.swift
import Foundation

/// Repository protocol for analytics and dashboard data.
protocol AnalyticsRepository {

    /// Fetches the main dashboard analytics summary.
    func getDashboardAnalytics() async throws -> DashboardAnalytics

    /// Fetches the incidents timeline for a given number of days.
    func getIncidentsTimeline(days: Int) async throws -> IncidentsTimeline

    /// Fetches the distribution of incidents by status, hazard type, and urgency.
    func getIncidentsDistribution() async throws -> IncidentsDistribution

    /// Fetches geographic analytics data for incident regions.
    func getGeographicAnalytics() async throws -> GeographicAnalytics

    /// Fetches all analytics data in a single call.
    func getAllAnalytics() async throws -> AnalyticsData
}
